test/ranges.jl

```diff
@@ -2112,3 +2112,16 @@ end
 @test length(range(1, 100, length=big(100)^100)) == big(100)^100
 @test length(range(big(1), big(100)^100, length=big(100)^100)) == big(100)^100
 @test length(0 * (1:big(100)^100)) == big(100)^100
+
+@testset "issue #41784" begin
+    # test `in` when step equals 0
+    # test for Int
+    x = abs(rand(Int))
+    @test (x in StepRangeLen(x, 0, abs(rand(Int)))) == true
```

Comment:
If the length is zero (probability is small but non-zero), this will fail.
```suggestion
    @test (x in StepRangeLen(x, 0, rand(1:100))) == true
```
instead?

Reply:
Ah, that's right too. I was trying to introduce as much randomness as I could but makes more sense!